ginivision/src/main/java/net/gini/android/vision/internal/camera/api/CameraController.java

```diff
@@ -336,24 +337,24 @@ private void configureCamera(Activity activity) {
 
         Camera.Parameters params = mCamera.getParameters();
 
-        List<Camera.Size> previewSizes = params.getSupportedPreviewSizes();
-        Size previewSize = getLargestFourThreeRatioSize(previewSizes);
-        if (previewSize != null) {
-            mPreviewSize = previewSize;
-            params.setPreviewSize(mPreviewSize.width, mPreviewSize.height);
-            LOG.debug("Preview size ({}, {})", mPreviewSize.width, mPreviewSize.height);
-        } else {
-            LOG.warn("No 4:3 preview size found");
-        }
-
         List<Camera.Size> pictureSizes = params.getSupportedPictureSizes();
-        Size pictureSize = getLargestFourThreeRatioSize(pictureSizes);
+        Size pictureSize = getLargestSize(pictureSizes);
         if (pictureSize != null) {
             mPictureSize = pictureSize;
             params.setPictureSize(mPictureSize.width, mPictureSize.height);
             LOG.debug("Picture size ({}, {})", mPictureSize.width, mPictureSize.height);
         } else {
-            LOG.warn("No 4:3 picture size found");
+            LOG.warn("No suitable picture size found");
+        }
+
+        List<Camera.Size> previewSizes = params.getSupportedPreviewSizes();
+        Size previewSize = getLargestSameAspectRatioSize(previewSizes, mPictureSize);
+        if (previewSize != null) {
+            mPreviewSize = previewSize;
+            params.setPreviewSize(mPreviewSize.width, mPreviewSize.height);
+            LOG.debug("Preview size ({}, {})", mPreviewSize.width, mPreviewSize.height);
+        } else {
+            LOG.warn("No suitable preview size found");
```

Comment:
Without the 4:3 limitation, going to this else would be a serious problem. It means the camera has no possible resolutions e.g. it doesn't work at all.

Will an exception be thrown elsewhere because of this? Because just a log message is probably not enough.